rpcserver.go

```diff
@@ -809,15 +814,151 @@ func (r *rpcServer) VerifyProof(ctx context.Context,
 	)
 	valid := err == nil
 
-	// TODO(roasbeef): also show additional final resting anchor
-	// information, etc?
+	rpcProof, err := marshalProofFile(proofFile, -1)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
 
-	// TODO(roasbeef): show the final resting place of the asset?
-	return &tarorpc.ProofVerifyResponse{
-		Valid: valid,
+	return &tarorpc.VerifyProofResponse{
+		Valid:        valid,
+		DecodedProof: rpcProof,
 	}, nil
 }
 
+// DecodeProof attempts to decode a given proof file that claims to be anchored
+// at the specified genesis point.
+func (r *rpcServer) DecodeProof(ctx context.Context,
+	in *tarorpc.DecodeProofRequest) (*tarorpc.DecodeProofResponse, error) {
+
+	if len(in.RawProof) == 0 {
+		return nil, fmt.Errorf("proof file must be specified")
+	}
+
+	var proofFile proof.File
+	err := proofFile.Decode(bytes.NewReader(in.RawProof))
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode proof file: %w", err)
+	}
+
+	resp, err := marshalProofFile(proofFile, in.ProofIndex)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
+
+	return resp, nil
+}
+
+// marshalProofFile turns a proof file into an RPC DecodeProofResponse.
+func marshalProofFile(proofFile proof.File, index int64) (*tarorpc.DecodeProofResponse, error) {
+	var (
+		decodedProof *proof.Proof
+		err          error
+	)
+
+	if index < 0 {
+		decodedProof, err = proofFile.LastProof()
+		if err != nil {
+			return nil, err
+		}
+		index = int64(proofFile.NumProofs() - 1)
+	} else {
+		decodedProof, err = proofFile.ProofAt(uint32(index))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var (
+		a                 = decodedProof.Asset
+		rpcAsset          = &tarorpc.Asset{}
+		assetID           = a.Genesis.ID()
+		anchorTxId        = decodedProof.AnchorTx.TxHash().String()
+		anchorOutputIndex = strconv.FormatUint(uint64(decodedProof.InclusionProof.OutputIndex), 10)
+		anchorOutpoint    = anchorTxId + ":" + anchorOutputIndex
+	)
+
+	var anchorTxBuf bytes.Buffer
```

Comment:
Can move this out into a serialization function similar to how things are above. Also could use some comments in the spirit of our lnd contribution guidelines: https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#code-documentation-and-commenting

Reply:
Should be cleaned up by leveraging the already-in-place marshalAsset()